logstash-core/lib/logstash/config/source/multi_local.rb

```diff
@@ -30,7 +30,7 @@ def initialize(settings)
     end
 
     def pipeline_configs
-      pipelines = deep_replace(retrieve_yaml_pipelines())
+      pipelines = retrieve_yaml_pipelines
```

Comment:
okay, things are clear now.
We do not _really_ need var substitution when loading the YAML. Once we have pipeline definitions from YAML, we  [properly load the configs](https://github.com/mashhurs/logstash/blob/main/logstash-core/lib/logstash/config/source/local.rb#L202-L216) based on the given options (`config.string`, `config.path`). Then, imperative pipeline compile operation executes [the LSCL](https://github.com/mashhurs/logstash/blob/b0948a78c01f5abc6183f9b37f037b50afe63cad/logstash-core/src/main/java/org/logstash/config/ir/ConfigCompiler.java#L94) which wipes out the comments based on the [rule](https://github.com/mashhurs/logstash/blob/7dde084ba66ea0572da26a6a673905202ec84ebb/logstash-core/lib/logstash/compiler/lscl/lscl_grammar.treetop#L9-L11)